theme/theme.inc

```diff
@@ -6,27 +6,72 @@
  */
 
 /**
- * Implements hook_preprocess_theme().
+ * Prepares variables for islandora_book_book templates.
+ *
+ * Default template: islandora-book-book.tpl.php.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - object: An AbstractObject for which to generate the display.
  */
-function islandora_book_preprocess_islandora_book_book(array &$variables) {
+function template_preprocess_islandora_book_book(array &$variables) {
   module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
+  module_load_include('inc', 'islandora', 'includes/solution_packs');
+
   $object = $variables['object'];
-  $params = array(
+
+  $variables['viewer_id'] = islandora_get_viewer_id('islandora_book_viewers');
+  $variables['viewer_params'] = array(
     'object' => $object,
     'pages' => islandora_paged_content_get_pages($object),
     'page_progression' => islandora_paged_content_get_page_progression($object),
   );
-  module_load_include('inc', 'islandora', 'includes/solution_packs');
-  $viewer = islandora_get_viewer($params, 'islandora_book_viewers');
-  if ($viewer) {
-    $variables['viewer'] = $viewer;
+
+  // @todo: Remove after 7.x-1.5 release.
+  $variables['viewer'] = new IslandoraBookDeprecateString($variables, 'viewer', 'template_process_islandora_book_book');
+}
+
+/**
+ * Processes variables for islandora_book_book templates.
+ *
+ * Default template: islandora-book-book.tpl.php.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - object: An AbstractObject for which to generate the display.
+ *   - viewer_id: The "ID" of the viewer to use.
+ *   - viewer_params: An (associative) array of parameters to pass to the
+ *     viewer.
+ */
+function template_process_islandora_book_book(array &$variables) {
+  // @todo: Remove condition with "IslandoraBookDeprecateString" after
+  // 7.x-1.5 release.
+  if ($variables['viewer'] instanceof IslandoraBookDeprecateString) {
+    if ($variables['viewer_id'] && $variables['viewer_id'] != 'none') {
+      $viewer_callback = islandora_get_viewer_callback($variables['viewer_id']);
+      if (function_exists($viewer_callback)) {
```

Comment:
Might need an else setting the var as null, maybe an islandora_call_viewer_callback func would be useful.